pkg/liquidity-mining/test/ChildChainGaugeRewardHelper.test.ts

```diff
@@ -0,0 +1,152 @@
+    balToken = await Token.create({ symbol: 'BAL' });
+
+    const gaugeImplementation = await deploy('RewardsOnlyGauge', {
+      args: [balToken.address, vault.address, adaptor.address],
+    });
+    const streamerImplementation = await deploy('ChildChainStreamer', { args: [balToken.address, adaptor.address] });
+
+    const factory = await deploy('ChildChainLiquidityGaugeFactory', {
+      args: [gaugeImplementation.address, streamerImplementation.address],
+    });
+
+    await factory.create(tokens.addresses[0]);
+
+    gaugeOne = await deployedAt('RewardsOnlyGauge', await factory.getPoolGauge(tokens.addresses[0]));
+    streamerOne = await deployedAt('ChildChainStreamer', await factory.getPoolStreamer(tokens.addresses[0]));
+
+    await factory.create(tokens.addresses[1]);
+    gaugeTwo = await deployedAt('RewardsOnlyGauge', await factory.getPoolGauge(tokens.addresses[1]));
+    streamerTwo = await deployedAt('ChildChainStreamer', await factory.getPoolStreamer(tokens.addresses[1]));
+
+    gaugeHelper = await deploy('ChildChainGaugeRewardHelper');
+  });
+
+  sharedBeforeEach('stake into gauge', async () => {
+    await tokens.mint({ to: user });
+    await tokens.approve([
+      { to: gaugeOne, from: user },
+      { to: gaugeTwo, from: user },
+    ]);
+
+    await gaugeOne.connect(user)['deposit(uint256)'](rewardAmount);
+    await gaugeOne.connect(user)['deposit(uint256,address)'](rewardAmount.mul(2), other.address);
+    await gaugeTwo.connect(user)['deposit(uint256)'](rewardAmount);
+    await gaugeTwo.connect(user)['deposit(uint256,address)'](rewardAmount.mul(2), other.address);
+  });
+
+  sharedBeforeEach('set up distributor on streamer', async () => {
+    const setDistributorActionId = await actionId(adaptor, 'set_reward_distributor', streamerOne.interface);
+    await vault.grantPermissionsGlobally([setDistributorActionId], admin);
+
+    const calldata = streamerOne.interface.encodeFunctionData('set_reward_distributor', [
+      balToken.address,
+      distributor.address,
+    ]);
+    await adaptor.connect(admin).performAction(streamerOne.address, calldata);
+    await adaptor.connect(admin).performAction(streamerTwo.address, calldata);
+  });
+
+  sharedBeforeEach('send tokens to streamer', async () => {
+    await balToken.mint(streamerOne.address, rewardAmount);
+    await streamerOne.connect(distributor).notify_reward_amount(balToken.address);
+
+    await balToken.mint(streamerTwo.address, rewardAmount);
+    await streamerTwo.connect(distributor).notify_reward_amount(balToken.address);
+  });
+
+  describe('claimRewardsFromGauge', () => {
+    context('when the gauge has not claimed tokens from streamer recently', () => {
+      it('claims tokens from the streamer', async () => {
+        const tx = await gaugeHelper.claimRewardsFromGauge(gaugeOne.address, user.address);
+        expectTransferEvent(await tx.wait(), { from: streamerOne.address, to: gaugeOne.address }, balToken);
```

Comment:
We can augment these tests by calling `pendingRewards` and checking that the amount transfered matches exactly the pending rewards.

We can also add tests where we check that `pendingRewards` increases with the expected rate as time goes by, even if there's been a recent claim.

Reply:
> We can augment these tests by calling pendingRewards and checking that the amount transfered matches exactly the pending rewards.

I think this is something which is covered by the tests for the `RewardsOnlyGauge`. It's the responsibility of that contract to properly pay out rewards once the tokens are sent to it, this contract just ensures those tokens have been sent.